pandas/core/strings.py

```diff
@@ -1737,9 +1757,17 @@ def cons_row(x):
                 return cons(result, name=name, index=index)
 
     @copy(str_cat)
-    def cat(self, others=None, sep=None, na_rep=None):
+    def cat(self, others=None, sep=None, na_rep=None, align=None):
+        from pandas.core.series import Series
+        # FutureWarning for align=None emitted in one place only: str_cat
+        if align is not None and align and isinstance(others, Series):
+            # str_cat deals with arrays only;
+            # make sure index is correct here as well for using _wrap_result
+            self._orig, others = self._orig.align(others, join='outer')
```

Comment:
why is this needed here? is this not handled in str_cat? (which dispatched to _get_array_list)?